operator/src/network/controller.rs

```diff
@@ -192,7 +194,8 @@ async fn reconcile(
             delete_job(cx.clone(), &ns, BOOTSTRAP_JOB_NAME).await?;
         } else {
             // We have fewer peers, remove the extra peers from the list
-            status.peers.retain(|peer| peer.index < spec.replicas);
+            // Note since all *extra* peer use negative indexes they should always be retained.
```

Comment:
didnt understand how the negative indexes are used?

Reply:
Negative indexes are not special. We need an index just to organize and provide an order to all peers in the network (at least from Keramik's perspective). We have two kinds of peers in the network, Ceramic peers and vanilla IPFS peers. For simplicity the implementation assigns indexes starting at 0 and increasing to Ceramic peers and assigns vanilla IPFS peers starting at -1 and decreasing. 

Again this is just an implementation detail but figured I'd call it out in the comments that are comparing indexes to peer counts.